src/network/websockets/server/types.ts

```diff
@@ -25,7 +25,10 @@ export type Params<AuthDetails> = {
       kind: string;
       id: string;
     }) => boolean;
-    parseToken: (token: string) => Promise<AuthDetails>;
+    parseToken: (params: {
```

Comment:
Forgive me for bike-shedding, but would the following shape work? When an operation has a single, clear required parameter, I think there’s value in keeping it separate to signify its importance.

```ts
parseToken: (token: string, params: {
      close: () => void;
}) => Promise<AuthDetails>;
```

Reply:
Sure, I like that approach.